app/controllers/application_controller.rb

```diff
@@ -1,10 +1,8 @@
 class ApplicationController < ActionController::API
 
-  rescue_from ActiveRecord::RecordNotFound, :with => :error_404
+  rescue_from ActiveRecord::RecordNotFound, with: ->{ show_errors(404) }
 
-  private
-
-  def error_404
-    head 404
+  def show_errors(status_code = params[:status_code] || 500)
+    head status_code
```

Comment:
This will cause all controllers (that inherit from this) to have a `show_errors` action.  Probably best avoided if possible.